package/MDAnalysis/analysis/msd.py

```diff
@@ -338,12 +359,18 @@ def __init__(self, u, select='all', msd_type='xyz', fft=True, **kwargs):
         # result
         self.results.msds_by_particle = None
         self.results.timeseries = None
+        if not fft and nongaussian:
+            self.results.nongaussian_by_particle = None
+            self.results.nongaussian_parameter = None
 
     def _prepare(self):
         # self.n_frames only available here
         # these need to be zeroed prior to each run() call
         self.results.msds_by_particle = np.zeros((self.n_frames,
                                                   self.n_particles))
+        if not self.fft and self.nongaussian:
```

Comment:
This should become one unified check on `nongaussian` if exeption is raised for `nongaussian = True and fft=True`